run_unittests.py

```diff
@@ -6733,6 +6733,31 @@ def test_build_rpath(self):
         install_rpath = get_rpath(os.path.join(self.installdir, 'usr/bin/progcxx'))
         self.assertEqual(install_rpath, 'baz')
 
+    @skipIfNoPkgconfig
+    def test_build_rpath_pkgconfig(self):
+        '''
+        Test that current build artefacts (libs) are found first on the rpath,
+        manually specified rpath comes second and additional rpath elements (from
+        pkg-config files) come last
+        '''
+        if is_cygwin():
+            raise unittest.SkipTest('Windows PE/COFF binaries do not use RPATH')
+        testdir = os.path.join(self.unit_test_dir, '89 pkgconfig build rpath order')
+        self.init(testdir, override_envvars={'PKG_CONFIG_PATH': testdir})
+        self.build()
+        # C program RPATH
+        build_rpath = get_rpath(os.path.join(self.builddir, 'prog'))
+        self.assertEqual(build_rpath, '$ORIGIN/sub:/foo/bar:/foo/dummy')
+        self.install()
+        install_rpath = get_rpath(os.path.join(self.installdir, 'usr/bin/prog'))
+        self.assertEqual(install_rpath, '/baz:/foo/dummy')
+        # C++ program RPATH
+        build_rpath = get_rpath(os.path.join(self.builddir, 'progcxx'))
+        self.assertEqual(build_rpath, '$ORIGIN/sub:/foo/bar:/foo/dummy')
+        self.install()
```

Comment:
You are running install twice. Once is sufficient. So something like:

- build
- check C build dir rpath
- check C++ build dir rpath
- install
- check C install rpath
- check C++ install rpath